lib/elixir/lib/calendar/iso.ex

```diff
@@ -820,6 +820,35 @@ defmodule Calendar.ISO do
     "~T[" <> Calendar.ISO.time_to_string(hour, minute, second, microsecond) <> "]"
   end
 
+  @doc """
+  Implements date parsing in support of `Kernel.sigil_D/2` for
+  this calendar
+  """
+  @doc since: "1.10.0"
+  @impl true
+  @spec parse_date!(String.t()) ::
+          {
+            Calendar.year(),
+            Calendar.month(),
+            Calendar.day()
+          }
+          | no_return()
```

Comment:
we should use `no_return` only if the function _never_ returns.

Also, since it's a callback I think we tend to just leave it with `@impl true` (and with neither `@doc` nor `@spec`) but not sure if that's a strict rule.